File: leetcode-algorithms/027. Remove Element/solution.py
def removeElement(nums, val):
	"""
	:type nums: List[int]
	:type val: int
	:rtype: int
	"""

	begin = 0
	for i in range(len(nums)):
		if nums[i] != val:
			nums[begin] = nums[i]
			begin += 1
	return begin



        

if __name__ == '__main__':
	nums = [3,2,2,3]
	val = 3
	res = removeElement(nums, val)
	print(res)

File: leetcode-algorithms/027. Remove Element/solution.cpp
#include <iostream>
#include <vector>
using namespace std;

int removeElement(vector<int>& nums, int val) {
	int begin = 0;
	for(int i=0;i<nums.size();i++)
	{
		if(nums[i]!=val)
			nums[begin++] = nums[i];
	}
	return begin;
}

int main()
{
	vector<int> nums = {3,2,2,3};
	int res = removeElement(nums,3);
	cout<<res<<endl;
	return 0;
}